react/Page/index.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Layout components that know how to handle the keyboard.
+ *
+ * Work when the webview does not resize when the keyboard appears.
+ *
+ * https://github.com/ionic-team/cordova-plugin-ionic-keyboard
+ * <preference name="KeyboardResize" value="false" />
+ */
+
+import { useKeyboardInfo } from './keyboard'
+import styles from './styles.styl'
+
+const TOP_BAR_HEIGHT = 48
+const BOTTOM_BAR_HEIGHT = 48
+
+/**
+ * Returns the min-height CSS property for the Page
+ *
+ * The goal is for the Page to fill all the screen that is visible to the user.
+ * Since the keyboard appearing does not have any effect on the layout of the
+ * page (), we have to remove its height from the available space when it appears.
+ *
+ * Also handles fixed bars space:
+ *
+ * - topBar height is removed from the real estate available.
+ * - bottomBar height is removed from the real estate available, unless the
+ *   keyboard is visible : since the bottom bar is expected to be hidden
+ *   when an input is focused (to prevent iOS flickers), it does not take space
+ *   when the keyboard is visible.
+ */
+const contentHeight = ({
+  topBarHeight = TOP_BAR_HEIGHT,
+  bottomBarHeight: BOTTOM_BAR_HEIGHT,
+  extraHeight = 0,
+  keyboardState,
+  keyboardHeight
+} = {}) => {
+  const removedSpace =
+    topBarHeight +
+    (keyboardState === 'showing' ? 0 : bottomBarHeight) +
+    // Surprisingly, vh changes when keyboard appears and 1 keyboard is added
+    // to the vh, this is why instead of adding 1 keyboardHeight, we add 2.
+    (keyboardState === 'showing' ? 2 * keyboardHeight : 0) +
+    extraHeight
+  return `calc(100vh ${removedSpace > 0 ? '-' : '+'} ${Math.abs(
+    removedSpace
+  )}px)`
+}
+
+/**
+ * Empty container, that has the same height as the keyboard when it is opened,
+ * used as a "cushion" upon which the PageFooter rests when the keyboard is opened.
+ */
+const KeyboardSpace = () => {
+  const { keyboardState, keyboardHeight } = useKeyboardInfo()
+  return (
+    <div
+      style={{
+        height: keyboardState === 'showing' ? keyboardHeight : 0
+      }}
+    >
+      {' '}
+    </div>
+  )
+}
+
+export const PageLayout = ({ children, extraHeight = 0 }) => {
+  const { keyboardState, keyboardHeight } = useKeyboardInfo()
```

Comment:
Can we add : 
if(!window.Keyboard){ console.warn('This component works with ...')} ? 

Also does it work on mobile web version also? Or do we need to do a check?

Reply:
- Added the warning in the "mount/update" phase of the effect.
 - On mobile web, the button will be behind the keyboard